lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -300,13 +300,34 @@ static Value commonShflSync(Location loc, ConversionPatternRewriter &rewriter,
 
 Value shflSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
                int i) {
-  return commonShflSync(loc, rewriter, val, i, "bfly", "0x1f");
+  return commonShflSync(loc, rewriter, val, i32_val(i), "bfly", "0x1f");
 }
 
 Value shflUpSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
                  int i) {
+  return commonShflSync(loc, rewriter, val, i32_val(i), "up", "0x0");
+}
+
+Value shflIdxSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
+                  int i) {
+  return commonShflSync(loc, rewriter, val, i32_val(i), "idx", "0x1f");
+}
+
+Value shflSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
+               Value i) {
+  return commonShflSync(loc, rewriter, val, i, "bfly", "0x1f");
+}
+
+Value shflUpSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
+                 Value i) {
   return commonShflSync(loc, rewriter, val, i, "up", "0x0");
 }
```

Comment:
We don't need those at the time? I think they are unlikely to be useful

Reply:
Sure, I'll remove those function definitions